app/dashboard/management/commands/activity_report.py

```diff
@@ -0,0 +1,170 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
+
+import argparse
+import csv
+import datetime
+import re
+import StringIO
+from itertools import imap
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from dashboard.models import Bounty, Tip
+from marketing.mails import send_mail
+
+DATE_FORMAT = '%Y/%m/%d'
+def valid_date(v):
+    try:
+        return datetime.datetime.strptime(v, DATE_FORMAT)
+    except ValueError:
+        raise argparse.ArgumentTypeError('%s is not a date in YYYY/MM/DD format' % v)
+
+GITHUB_REPO_PATTERN = re.compile('github.com/[\w-]+/([\w-]+)')
+
+def extract_github_repo(url):
+    match = GITHUB_REPO_PATTERN.search(url)
+    if not match:
+        raise Exception("malformed github url: %s", url)
+
+    return match.groups()[0]
+
+def format_bounty(bounty):
+    return {
+        'type': 'bounty',
+        'created_on': bounty.created_on,
+        'last_activity': bounty.modified_on,
+        'amount': bounty.value_in_token,
+        'denomnation': bounty.token_name,
+        'amount_eth': bounty.value_in_eth,
+        'amount_usdt': bounty.value_in_usdt,
+        'from_address': bounty.bounty_owner_address,
+        'to_address': bounty.claimee_address,
+        'repo': extract_github_repo(bounty.github_url),
+        'from_username': bounty.bounty_owner_github_username or '', # XXX owner can be null
+        'to_username': bounty.claimee_github_username or '', # XXX claimee can be null
+        'status': bounty.status,
+        'comments': 'no bounty comments right now' # XXX where does the PR link come from?
+    }
+
+def format_tip(tip):
+    return {
+        'type': 'tip',
+        'created_on': tip.created_on,
+        'last_activity': tip.modified_on,
+        'amount': tip.amount,
+        'denomnation': tip.tokenName,
+        'amount_eth': tip.value_in_eth,
+        'amount_usdt': tip.value_in_usdt,
+        'from_address': 'tip has no owner address saved', # XXX owner address field can't be populated with current schema
+        'to_address': 'tip has no claimee', # XXX claimee address field doesn't make sense for tips
+        'repo': extract_github_repo(tip.github_url) if tip.github_url else '', # XXX url can be null
+        'from_username': tip.from_email, # XXX github owner field doesn't make sense for tips
+        'to_username': 'tip has no claimee', # XXX github claimee field doesn't make sense for tips
+        'status': tip.status,
+        'comments': 'no no tip comments right now' # XXX comment field doesn't make sense for tips
+    }
+
+def itermerge(gen_a, gen_b, key):
```

Comment:
i wonder if this belongs in some utils.py file somewhere..

Reply:
is it pasted from somewhere?  probably best to leave credit if so

Reply:
nah i wrote it because i didnt want to deal with resolving large querysets - i'll move to utils